alpharaw/mzml_io/mzml_writer.py

```diff
@@ -0,0 +1,840 @@
+import xml.etree.ElementTree as ET
+import base64
+import struct
+import os
+import zlib
+import hashlib
+import multiprocessing as mp
+import platform
+from functools import partial
+from tqdm import tqdm
+import numpy as np
+
+def _create_spectrum_xml(i, row, ms_data, binary_precision, compression, ns_prefix):
+    """Create XML string for a single spectrum"""
+    # Build spectrum using ElementTree
+    spectrum = ET.Element(ns_prefix + "spectrum")
+    
+    # Required attributes
+    spectrum.set("index", str(i))
+    spectrum.set("id", f"scan={i}")
+    
+    # Get peak data using position i
+    try:
+        mz_array, intensity_array = ms_data.get_peaks(i)
+    except Exception:
+        # If there's an error, create empty arrays
+        mz_array = np.array([])
+        intensity_array = np.array([])
+        
+    spectrum.set("defaultArrayLength", str(len(mz_array)))
+    
+    # MS level
+    ms_level = row.get('ms_level', 1)
+    cv_param = ET.SubElement(spectrum, ns_prefix + "cvParam")
+    cv_param.set("cvRef", "MS")
+    cv_param.set("accession", "MS:1000511")
```

Comment:
All of the controlled vocabulary is defined here:
https://github.com/HUPO-PSI/psi-ms-CV/blob/master/psi-ms.obo

Could we reuse this in some way or at least put it in a constants file:

```python
CV_REF = "cvRef"
ACCESSION_CENTROIDED = "MS:1000127"
ACCESSION_PROFILE = "MS:1000128"
```